checker/src/main/java/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -169,6 +178,47 @@ private boolean isHandledArrayCreation(Tree argument, String receiver) {
         return false;
     }
 
+    /**
+     * Returns true iff {@code arg} is a reference to a {@code private final} array field whose
+     * initializer is `new T[0]`, `new T[] {}`, or `{}`.
+     */
+    private boolean isZeroLengthPrivateFinalFieldArray(ExpressionTree arg) {
+
+        if (!(arg instanceof IdentifierTree) && !(arg instanceof MemberSelectTree)) {
+            return false;
+        }
+
+        Element el = TreeUtils.elementFromUse(arg);
+        if (el == null || !el.getKind().isField()) {
+            return false;
+        }
+        VariableElement var = (VariableElement) el;
+        if (!(var.getModifiers().containsAll(EnumSet.of(Modifier.PRIVATE, Modifier.FINAL))
+                && ElementUtils.getType(var).getKind() == TypeKind.ARRAY)) {
+            return false;
+        }
+
+        Trees trees = Trees.instance(atypeFactory.getProcessingEnv());
+        TreePath path = trees.getPath(var);
```

Comment:
Can you use https://github.com/eisop/checker-framework/blob/a757e144359cd6028c5ce5aa94633347abeec72b/framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java#L3967 to possibly use a cached path?